File: internal/modules/system/dto/casbin.go
package dto

// CasbinInfo Casbin 规则详情
type CasbinInfo struct {
	Path   string `json:"path"`   // API 路径
	Method string `json:"method"` // 请求方法
}

// UpdateCasbinReq 更新角色API权限请求
type UpdateCasbinReq struct {
	AuthorityId string       `json:"authorityId" binding:"required"` // 角色ID
	CasbinInfos []CasbinInfo `json:"casbinInfos"`                    // 选中的 API 列表
}

// GetPolicyPathByAuthorityIdReq 获取角色当前拥有的API权限
type GetPolicyPathByAuthorityIdReq struct {
	AuthorityId string `json:"authorityId" binding:"required"`
}

File: front-end/src/services/api/casbin.ts
// src/services/api/casbin.ts
import { request } from '@umijs/max';

// 更新 Casbin 权限
export async function updateCasbin(body: { authorityId: string, casbinInfos: { path: string, method: string }[] }) {
  return request<API.CommonResponse>('/api/v1/sys/casbin/updateCasbin', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    data: body,
  });
}

// 获取角色当前的 Casbin 权限
export async function getPolicyPathByAuthorityId(body: { authorityId: string }) {
  return request<API.CommonResponse>('/api/v1/sys/casbin/getPolicyPathByAuthorityId', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    data: body,
  });
}
